tasks/update_go.py

```diff
@@ -0,0 +1,179 @@
+import re
+from typing import Optional
+
+from invoke import Context, exceptions, task
+
+from .go import tidy_all
+from .libs.common.color import color_message
+from .modules import DEFAULT_MODULES
+from .pipeline import update_circleci_config, update_gitlab_config
+
+GO_VERSION_FILE = "./.go-version"
+
+
+# replace the given pattern with the given string in the file
+def _update_file(path: str, pattern: str, replace: str, expected_match: int = 1):
+    # newline='' keeps the file's newline character(s)
+    # meaning it keeps '\n' for most files and '\r\n' for windows specific files
+
+    with open(path, "r", newline='') as reader:
+        content = reader.read()
+
+    content, nb_match = re.subn(pattern, replace, content, flags=re.MULTILINE)
+    if nb_match != expected_match:
+        print(
+            color_message(
```

Comment:
What do you think about exiting the task instead as it is easy to miss a warning? You can also add a config flag to keep the possibility to ignore the error.

Reply:
I guess it would make sense